astropy/coordinates/baseframe.py

```diff
@@ -1201,6 +1201,12 @@ def transform_to(self, new_frame):
                                       'please comment at https://github.com/astropy/astropy/issues/6280')
 
         if inspect.isclass(new_frame):
+            warnings.warn("Transforming a frame instance to a frame class (as opposed to another "
+                          "frame instance) will not be supported in the future.  Either "
+                          "explicitly instantiate the target frame, or first convert the source "
+                          "frame instance to a `astropy.coordinates.SkyCoord` and use its "
+                          "`transform_to()` method.",
+                          AstropyDeprecationWarning)
```

Comment:
Good! Let's also remember to make an issue to remove this warning and functionality.